updater/library/src/config.rs

```diff
@@ -0,0 +1,68 @@
+// This file handles the global config for the updater library.
+
+use std::cell::RefCell;
+
+use crate::updater::AppConfig;
+
+const DEFAULT_BASE_URL: &'static str = "https://shorebird-code-push-api-cypqazu4da-uc.a.run.app";
+const DEFAULT_CHANNEL: &'static str = "stable";
+
+thread_local!(static CONFIG: RefCell<Option<ResolvedConfig>> = RefCell::new(None));
+
+pub fn with_config<F, R>(f: F) -> R
+where
+    F: FnOnce(&ResolvedConfig) -> R,
+{
+    CONFIG
+        .try_with(|config| {
```

Comment:
"with" or "try_with" is how you grab at thread locals in rust. 🤷   I'm using try_with here so I can have a custom crash message.  .expect says "if this was not OK, panic e.g. crash with this message".